app/os/main.cpp

```diff
@@ -225,38 +225,32 @@ namespace motors {
 
 }	// namespace motors
 
-namespace display {
+namespace display::internal {
 
-	namespace internal {
+	auto event_flags = CoreEventFlags {};
 
-		auto event_flags = CoreEventFlags {};
-
-		auto corell		   = CoreLL {};
-		auto pixel		   = CGPixel {corell};
-		auto hal		   = CoreSTM32Hal {};
-		auto coresdram	   = CoreSDRAM {hal};
-		auto coredma2d	   = CoreDMA2D {hal};
-		auto coredsi	   = CoreDSI {hal};
-		auto coreltdc	   = CoreLTDC {hal};
-		auto coregraphics  = CoreGraphics {coredma2d};
-		auto corefont	   = CoreFont {pixel};
-		auto coreotm	   = CoreLCDDriverOTM8009A {coredsi, PinName::SCREEN_BACKLIGHT_PWM};
-		auto corelcd	   = CoreLCD {coreotm};
-		auto _corejpegmode = CoreJPEGModeDMA {hal};
-		auto corejpeg	   = CoreJPEG {hal, _corejpegmode};
-		auto corevideo =
-			CoreVideo {hal, coresdram, coredma2d, coredsi, coreltdc, corelcd, coregraphics, corefont, corejpeg};
-
-		HAL_VIDEO_DECLARE_IRQ_HANDLERS(corevideo);
-
-	}	// namespace internal
+	auto corell		   = CoreLL {};
+	auto pixel		   = CGPixel {corell};
+	auto hal		   = CoreSTM32Hal {};
+	auto coresdram	   = CoreSDRAM {hal};
+	auto coredma2d	   = CoreDMA2D {hal};
+	auto coredsi	   = CoreDSI {hal};
+	auto coreltdc	   = CoreLTDC {hal};
+	auto coregraphics  = CoreGraphics {coredma2d};
+	auto corefont	   = CoreFont {pixel};
+	auto coreotm	   = CoreLCDDriverOTM8009A {coredsi, PinName::SCREEN_BACKLIGHT_PWM};
+	auto corelcd	   = CoreLCD {coreotm};
+	auto _corejpegmode = CoreJPEGModeDMA {hal};
+	auto corejpeg	   = CoreJPEG {hal, _corejpegmode};
 
-	auto videokit = VideoKit {internal::event_flags, internal::corevideo};
+	extern "C" auto corevideo =
+		CoreVideo {hal, coresdram, coredma2d, coredsi, coreltdc, corelcd, coregraphics, corefont, corejpeg};
 
-}	// namespace display
+}	// namespace display::internal
 
-auto behaviorkit   = BehaviorKit {display::videokit, leds::kit, motors::left::motor, motors::right::motor};
-auto reinforcerkit = ReinforcerKit {display::videokit, leds::kit, motors::left::motor, motors::right::motor};
+auto videokit	   = VideoKit {display::internal::event_flags, display::internal::corevideo};
+auto behaviorkit   = BehaviorKit {videokit, ledkit, motors::left::motor, motors::right::motor};
+auto reinforcerkit = ReinforcerKit {videokit, ledkit, motors::left::motor, motors::right::motor};
```

Comment:
it could but adds more changes.
if you really insist, but I feel it's not very important here. 
if we do it here, we need to do it for other places where the same namespace structure is used, which will add more changes.

I think a follow up PR focused on this issue would be better